models/hashflow/avalanche_c/hashflow_avalanche_c_crosschain_trades.sql

```diff
@@ -0,0 +1,103 @@
+                  WHEN dstChainId = 9 OR dstChainId = 109 THEN 'Polygon'
+                  WHEN dstChainId = 2 OR dstChainId = 102 THEN 'BNB' END AS destination_chain
+            ,'Avalanche'                   AS source_chain
+        FROM
+            {{ source('hashflow_avalanche_c', 'Pool_evt_LzTrade') }}
+        {% if is_incremental() %}
+            WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+        
+        UNION
+
+        SELECT
+            evt_block_time           AS block_time
+            ,trader            
+            ,quoteTokenAmount        AS token_bought_amount_raw
+            ,baseTokenAmount         AS token_sold_amount_raw
+            ,cast(NULL AS double)    AS amount_usd
+            ,quoteToken              AS token_bought_address
+            ,baseToken               AS token_sold_address
+            ,evt_tx_hash             AS tx_hash
+            ,CASE WHEN dstChainId = 1 THEN 'Ethereum'
+                  WHEN dstChainId = 2 THEN 'Arbitrum'
+                  WHEN dstChainId = 3 THEN 'Optimism'
+                  WHEN dstChainId = 4 THEN 'Avalanche'
+                  WHEN dstChainId = 5 THEN 'Polygon'
+                  WHEN dstChainId = 6 THEN 'BNB' END AS destination_chain
+            ,'Avalanche'                   AS source_chain
+        FROM
+            {{ source('hashflow_avalanche_c', 'Pool_evt_XChainTrade') }}
+        {% if is_incremental() %}
+            WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+)
+
+SELECT
+    try_cast(date_trunc('DAY', cross_chain_trades.block_time) AS date)       AS block_date
+    ,cross_chain_trades.block_time
+    ,erc20a.symbol                                                           AS token_bought_symbol
+    ,erc20b.symbol                                                           AS token_sold_symbol
+    ,cross_chain_trades.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount
+    ,cross_chain_trades.token_sold_amount_raw / power(10, erc20b.decimals)   AS token_sold_amount
+    ,CAST(cross_chain_trades.token_bought_amount_raw AS DECIMAL(38,0))       AS token_bought_amount_raw
+    ,CAST(cross_chain_trades.token_sold_amount_raw AS DECIMAL(38,0))         AS token_sold_amount_raw
+    ,coalesce(
+            cross_chain_trades.amount_usd
+        , (cross_chain_trades.token_bought_amount_raw / power(10, p_bought.decimals)) * p_bought.price
+        , (cross_chain_trades.token_sold_amount_raw / power(10, p_sold.decimals)) * p_sold.price
+        )                                                                   AS amount_usd
+    ,cross_chain_trades.token_bought_address
+    ,cross_chain_trades.token_sold_address
+    ,cross_chain_trades.trader
+    ,cross_chain_trades.tx_hash
+    ,cross_chain_trades.source_chain
+    ,cross_chain_trades.destination_chain
+FROM cross_chain_trades
+LEFT JOIN {{ ref('tokens_erc20') }} erc20a
```

Comment:
you may want to add join conditions on `tokens_erc20` throughout for `blockchain = <current blockchain>`, or you could get accidental duplicate returns

Reply:
I tried this initially but realized that it was creating a lot of NULLs and missing data because of the nature of the trade events. All trades from these events are cross chain trades so the blockchains and token addresses are always different/changing depending on what chains the user is swapping to, am unable to specify a single chain to join on here without losing data.

Reply:
Apologies for the delay, yep I added the join on erc20b